tox.ini

```diff
@@ -37,6 +37,8 @@ deps =
      # regressions in new releases of this package that cause us the kind of
      # suffering we're trying to avoid with the above pins.
      certifi
+     # VCS hooks support
+     py36,!coverage: pre-commit
```

Comment:
Do we really need `pre-commit` installed in the tox environment as well? (Or, maybe I don't know what this is saying).

Reply:
It's a pragmatic convenience.  The testing tox venvs already include other developer tools and as such don't religiously reproduce deployment environments.  IMO, the contents of the `extras_require` in `./setup.py` should be cleaned up as there seem to be dependencies among them that are optional niceties and not actually required for installation, usage, or development, but I would think that belongs in another PR if tackled.  So this isn't a step backward on that count, the risk of adding `pre-commit` is minimal, and since this is for a documented developer usage it's better than many of the others.

Reply:
BTW, I just ran into a somewhat related issue with the results from merging #798 while working on another branch.  Because the hook from that PR is *not* using the tox venvs, it's using `flake8` under Python 3 from my local host OS install which fails appropriately in this still Python 2.7 code base:
```
Pushing to git@github.com:tahoe-lafs/tahoe-lafs.git
[INFO] Initializing environment for https://gitlab.com/pycqa/flake8.
[INFO] Installing environment for https://gitlab.com/pycqa/flake8.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
flake8...................................................................Failed
- hook id: flake8
- exit code: 1

src/allmydata/test/mutable/test_version.py:338:22: F821 undefined name 'xrange'
src/allmydata/web/common.py:136:21: F821 undefined name 'unicode'
src/allmydata/web/common.py:198:12: F821 undefined name 'unicode'
src/allmydata/web/common.py:329:29: F821 undefined name 'unicode'
src/allmydata/test/common_util.py:21:32: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:486:28: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:490:24: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:573:34: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2575:34: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2582:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2589:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2597:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2606:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2615:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2628:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2639:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2652:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2831:33: F821 undefined name 'unicode'
src/allmydata/test/web/test_web.py:2930:32: F821 undefined name 'unicode'

error: failed to push some refs to 'git@github.com:tahoe-lafs/tahoe-lafs.git'
```

This is an example of the kind of thing that's avoided by keeping development tools withing the project's tox venvs.